types/convert.go

```diff
@@ -378,6 +378,21 @@ func ConvertJSONToFloat(sc *stmtctx.StatementContext, j json.BinaryJSON) (float6
 	return 0, errors.New("Unknown type code in JSON")
 }
 
+// ConvertJSONToDecimal casts JSON into decimal.
+func ConvertJSONToDecimal(sc *stmtctx.StatementContext, j json.BinaryJSON) (*MyDecimal, error) {
+	res := new(MyDecimal)
+	if j.TypeCode != json.TypeCodeString {
+		f64, err := ConvertJSONToFloat(sc, j)
+		if err != nil {
+			return res, errors.Trace(err)
+		}
+		err = res.FromFloat64(f64)
+		return res, errors.Trace(err)
```

Comment:
What about create two new function `NewDecFromFloat` and `NewDecFromString` and call it here?

Reply:
`MyDecimal` already have `FromFloat64` and `FromString` methods.  What is the difference between `NewDecFromFloat` and `FromFloat64`?